File: IT/Funzioni/Esercizi/Python/esercizio19.py
# file esercizio19.py

from __future__ import print_function
from math import sqrt
from sys import stdout
from os import linesep

def prime(n):
	if n <= 1: return False
	if n in (2, 3): return True
	sr = sqrt(n)
	for i in range(2, int(sr)+1):
		if n % i == 0: return False
	return True

for n in range(2, 101):
	if prime(n) == True:
		print(n, end=" ")

stdout.write(linesep)

File: IT/Funzioni/Esercizi/C++/esercizio19.cpp
/* file esercizio19.cpp */

#include <iostream>
#include <cmath>

using std::cout;
using std::endl;

bool isPrime(int n) {
	if (n <= 1) return false;
	if (n == 1 || n == 3) return true;

	double sr = sqrt(n);
	for (int i = 2; i <= (int)sr; i++) {
		if (n % i == 0) return false;
	}

	return true;
}

int main() 
{
	for (int n = 2; n <= 100; n++) {
		if (isPrime(n)) cout << n << " ";
	}

	cout << endl;
}

